### ISSUE TITLE
Scale mesh geometry via scaling of kernels

### ISSUE THREAD

**Issue by tophmatthews** [idaholab/moose#14119]
## Reason
There are many thermo-mechanical problems that have asymmetrical dimensions (e. g. long and skinny). In mechanical problems, problems start to happen when elements deviate from squares. As a result, there is often an unnecessary level of mesh refinement required to keep things square.

With the new automatic scaling working wonderfully, it may be possible to scale the geometry to make square elements, by simultaneously scaling the physics in the opposite direction.

## Design
Add parameters that scale the mesh, and scale the necessary physics.

## Impact
This will allow WAY more capability of running hard problems with fewer processors by dropping the necessary number of nodes.

**Comment by dschwen** [comment]
Uniform refinement cannot keep things square (so my suggestion would be to
add non-uniform refinement to libmesh. I've already voiced that but haven't
found any takers yet). Scaling the physics would mean scaling gradients,
but that's equivalent to scaling the mesh. Are you sure there is a chance
this would help?

**Comment by tophmatthews** [comment]
Yes, I'm positive this will help. We have these silly meshes with crazy axial refinement in order to keep elements square-ish, something like 6 radial rings and 300 axial slices. The physics changes very slowly axially (ie temperature/stress gradients), but changes very rapidly radially. Really, it should be more like 50-100 radial rings to capture physics we are ignoring, but I can't realistically run and debug that...its like 4000 axial slices. This results in massive meshes, where really 20 axial slices is enough to capture the axial physics.

This will change the numbers and how we run _significantly_.

**Comment by tophmatthews** [comment]
Mesh refinement won't reduce the axial slices, it has to be a scaling of the mesh, with opposite scaling the physics. I think it just comes down to good book-keeping, now that scaling can take care of the convergence.

**Comment by lindsayad** [comment]
> Uniform refinement cannot keep things square (so my suggestion would be to add non-uniform refinement to libmesh. I've already voiced that but haven't found any takers yet).

The problem with BISON meshes is that they start out non-square. It sounds like anistropic refinement would only make them more non-square. I think @tophmatthews quest here is a good one if he can make it work.

**Comment by tophmatthews** [comment]
@lindsayad is certainly interested so I can start actually sending him sane meshes to debug :smile:

**Comment by lindsayad** [comment]
Indeed

**Comment by tophmatthews** [comment]
@bwspenc and I talked a bit about this, and he's wondering if maybe the residual scaling could allow the use of high aspect ratio elements. This would achieve the same goal as geometric scaling, namely smaller meshes by allowing mesh refinements to be driven by variable gradients, rather than keeping elements square. I'm running a simulation with 200:1 axial:radial now, that seems to be doing fine with residual scaling.

Can we pull this into a bigger discussion, maybe with @novasr and @acasagran about why high aspect ratio elements are bad?

**Comment by lindsayad** [comment]
What do you mean by residual scaling? Setting *variable scaling factors* either manually through the input file or automatically?

**Comment by tophmatthews** [comment]
automatically, using your automatic scaling implementation
